src/shadowbox/integration_test/test.sh

```diff
@@ -79,10 +86,59 @@ function fail() {
   exit 1
 }
 
+function setup() {
+  shutdown_containers
+
+  docker network create -d bridge "${NET_OPEN}"
+  docker network create -d bridge --internal "${NET_BLOCKED}"
+
+  # Target service.
+  docker build --force-rm -t "${TARGET_IMAGE}" "$(dirname "$0")/target"
+  docker run -d --rm -p "10080:80" --network="${NET_OPEN}" --network-alias="target" --name="${TARGET_CONTAINER}" "${TARGET_IMAGE}"
+  
+  # Shadowsocks service.
+  declare -ar shadowbox_flags=(
+    -d
+    --rm
+    --network="${NET_BLOCKED}"
+    --network-alias="shadowbox"
+    -p "20443:443"
+    -e "SB_API_PORT=443"
+    -e "SB_API_PREFIX=${SB_API_PREFIX}"
+    -e "LOG_LEVEL=debug"
+    -e "SB_CERTIFICATE_FILE=/root/shadowbox/test.crt"
+    -e "SB_PRIVATE_KEY_FILE=/root/shadowbox/test.key"
+    -v "${SB_CERTIFICATE_FILE}:/root/shadowbox/test.crt"
+    -v "${SB_PRIVATE_KEY_FILE}:/root/shadowbox/test.key"
+    -v "${TMP_STATE_DIR}:/root/shadowbox/persisted-state"
+    --name "${SHADOWBOX_CONTAINER}"
+    "${SHADOWBOX_IMAGE}"
+  )
+  docker run "${shadowbox_flags[@]}"
+  # docker network connect --alias shadowbox "${NET_BLOCKED}" "${SHADOWBOX_CONTAINER}"
+  docker network connect "${NET_OPEN}" "${SHADOWBOX_CONTAINER}"
+
+  # Client service.
+  docker build --force-rm -t "${CLIENT_IMAGE}" "$(dirname "$0")/client"
+  # Use -i to keep the container running.
+  docker run -d --rm -it -p "30555:555" --network "${NET_BLOCKED}" --name "${CLIENT_CONTAINER}" "${CLIENT_IMAGE}"
+
+  # Utilities
+  docker build --force-rm -t "${UTIL_IMAGE}" "$(dirname "$0")/util"
+}
+
+function shutdown_containers() {
```

Comment:
This isn't just shutting down containers, it's also removing them. Maybe rename this function to reflect that or at least leave a comment for people less familiar with what `docker rm -f` does?